### PR TITLE
Add NarrowTypes

### LINKED ISSUE #75: Add `NarrowTypes` transform that converts columns to the most specific types possible
Add a transform that takes a table with possibly generic column eltypes and returns a new table with specific column types.

### PR DESCRIPTION
**Pull request by eliascarv**
This PR implements NarrowTypes transform.

closes #75

### CODE REVIEW

File: src/TableTransforms.jl [hunk 1]
```diff
@@ -39,6 +39,7 @@ export
   Coerce,
   Levels,
   OneHot,
+  NarrowTypes,
```
Comment (juliohm) [inline src/TableTransforms.jl:42]: Let's move this transform to before Coalesce

**Review by juliohm — CHANGES_REQUESTED** [review]
🚀

**Comment by eliascarv** [comment]
This PR depends on fixing the following issue from the Transducers.jl package:
https://github.com/JuliaFolds/Transducers.jl/issues/524

**Comment by juliohm** [comment]
Closing as discussed in #75

**Pull request closed without merging**